supervisor/iptables_test.go

```diff
@@ -105,3 +107,103 @@ func TestSuperviseACLs(t *testing.T) {
 	}
 
 }
+
+
+//Call Supervise and we will mock AddContainer Chain here  by changing the mock defintions of newChain
+//The function should cleanup on all errors 
+func TestAddContainerChain(t *testing.T){
+	s := doNewIPTSupervisor(t)
+	containerInfo := policy.NewPUInfo("12345")
+	containerInfo.Policy.IngressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+
+	containerInfo.Policy.EgressACLs = []policy.IPRule{policy.IPRule{
+		Address:  "20.20.0.0/16",
+		Port:     "80",
+		Protocol: "tcp",
+	}}
+	containerInfo.Runtime.SetIPAddresses(map[string]string{"bridge": "30.30.30.30"})
+	containerInfo.Policy.PolicyIPs = []string{"30.30.30.30"}
+	m := s.ipt.(TestIptablesProvider)
+
+	m.MockNewChain(t,func(table,chain string)(error){
+		if(table == "raw" ){
```

Comment:
Why are you specifically checking if the Table is raw before returning an error ? 
If I understand your test ace correctly you want to emulate an error when calling NewChain, isn't ?